### PR TITLE
fixes 3826: no more sql errors when trying to search for environments (also domains, operatingsystems, and models)

### PR DESCRIPTION
**Pull request by dmitri-d**

### CODE REVIEW

File: app/controllers/operatingsystems_controller.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@ class OperatingsystemsController < ApplicationController
 
   def index
     @operatingsystems = Operatingsystem.search_for(params[:search], :order => params[:order]).paginate(:page => params[:page])
-    @counter = Host.group(:operatingsystem_id).where(:operatingsystem_id => @operatingsystems.pluck(:id)).count
+    @counter = Host.group(:operatingsystem_id).where(:operatingsystem_id => @operatingsystems.collect(&:id)).count
```
Comment (lzap) [inline app/controllers/operatingsystems_controller.rb:7]: LGTM but why collect here and not there?

Comment (dmitri-d) [inline app/controllers/operatingsystems_controller.rb:7]: Not sure what you mean exactly. If you are asking about pluck vs. collect -- pluck causes issues on older ActiveRecord (3.2.8) and postgres when used in subselects. But then @operatingsystems is being used locally anyway, I'm not sure there's any point in using a subselect here (and other queries that needed fixes) in the first place.

Comment (lzap) [inline app/controllers/operatingsystems_controller.rb:7]: Oh okay. LGTM

## 

Later,

 Lukas "lzap" Zapletal
 irc: lzap #theforeman

**Comment by domcleal** [comment]
Thanks @witlessbird, merged as 145db86 for Foreman 1.4.0.

**Pull request closed without merging**